### ISSUE TITLE
Spot unable to connect the QrCode generated by Kotlin/Swift Dapp

### ISSUE THREAD

**Issue by kecson** [WalletConnect/WalletConnectKotlinV2#696]
**Describe the bug**

**SDK Version**
 - Client: [Kotlin]

**To Reproduce**
1. Open  `Spot` App. https://www.spot-wallet.com
2. Click Scan QrCode
3. 
![img_v2_5533fc86-ab08-4687-a094-09952199528h](https://user-images.githubusercontent.com/10434414/223306113-ddc139f7-653c-4bd8-a8a6-e3e48455c8d5.png)

**Comment by china-mobile2008** [comment]
@kecson 
May I ask if your Android dApp has slow connection to BitKeep wallet? Also, does it have a very low success rate and get stuck for a long time without any progress?

**Comment by jakubuid** [comment]
@kecson 
Did you use https://github.com/WalletConnect/WalletConnectKotlinV2/tree/develop/sign/dapp to connect to SpotWallet?
Did you do it by deep link flow or by scanning a QR from a sample Dapp?

**Comment by jakubuid** [comment]
During my testing, I noticed that the Spot wallet is not able to establish a connection with the example Kotlin Dapp either by deep link flow not by scanning the QR code

![Screenshot_20230309-061506](https://user-images.githubusercontent.com/23366251/223927547-5fae2b94-8753-40da-982a-352c28971935.png)

**Comment by CyrilNb** [comment]
Hello, Spot android developer here.

What's the URL of the example Kotlin Dapp we tare alking here, please? So I can take a look at it.
Thanks

**Comment by kecson** [comment]
> @kecson
> Did you use https://github.com/WalletConnect/WalletConnectKotlinV2/tree/develop/sign/dapp to connect to SpotWallet?
> Did you do it by deep link flow or by scanning a QR from a sample Dapp?

yes

**Comment by kecson** [comment]
> Hello, Spot android developer here.
> 
> What's the URL of the example Kotlin Dapp we tare alking here, please? So I can take a look at it. Thanks

Dapp sample code: https://github.com/WalletConnect/WalletConnectKotlinV2/tree/develop/sign/dapp

**Comment by jakubuid** [comment]
Hi, users reported that they cannot connect to: https://github.com/WalletConnect/WalletConnectKotlinV2/tree/develop/sign/dapp
which I was able to reproduce. After scanning the qr code from this Dapp the session proposal dialog is shown on the Spot wallet, however after pressing the Connect button the error screen is shown.

Could you have a look and check what's going on Spot? 

![Image](https://user-images.githubusercontent.com/23366251/224990422-ae8333a2-d3cd-40a2-88c4-da97a68aeb69.jpg)
![Image](https://user-images.githubusercontent.com/23366251/224990421-a3c0fce6-6fbb-4c00-aa83-99dd98459140.jpg)

 
Dapp doesn't any response from Spot.

**Comment by china-mobile2008** [comment]
@CyrilNb 
Please configure the project ID for the project.

For the Project ID look at [Project ID](https://www.walletconnect.com/).

**Comment by jakubuid** [comment]
@CyrilNb you can use https://drive.google.com/file/d/1wuenKAXhUdFfGlmmwSvurkqc8Iwk9AdK/view?usp=share_link
Lmk if you can fetch it

**Comment by CyrilNb** [comment]
thx Jakub, I finally managed to launch the kotlin dApp and reproduced the error. It's about InvalidNamespaceException, we're on it.

Do you know why it does not occur with your dApps tester like [lab.web3modal.com](https://lab.web3modal.com/) or https://react-app.walletconnect.com/ ?

**Comment by jakubuid** [comment]
Are you able to retrieve the error message included in InvalidNamespaceException?

**Comment by jakubuid** [comment]
These are the namespaces send from Kotlin Dapp when choosing only Ethereum Main:

```
{
  "requiredNamespaces": {
    "eip155": {
      "chains": [
        "eip155:1"
      ],
      "methods": [
        "eth_sendTransaction",
        "personal_sign",
        "eth_sign",
        "eth_signTypedData"
      ],
      "events": [
        "chainChanged",
        "accountChanged"
      ]
    }
  },
  "optionalNamespaces": {}
}
```

@CyrilNb if you could send me the response that is supposed to be send from Spot wallet in this case, I'll be able to help and see why the validation is not passing your response

Btw. what version of SDK is Spot using?

**Comment by CyrilNb** [comment]
Our backend module had already fixed the namespaces problem, it was not yet in production for Android.

So I am now testing it with the fix, no more InvalidNamespaceException but when sending the following `Wallet.Params.SessionApprove` object :

`SessionApprove(proposerPublicKey=b32b3eda7493b5d838aec95f8848126765812eaafdcb8a5593f0f65000df9046, namespaces={eip155=Session(accounts=[eip155:1:0xbEFB2814e6f503b9bD75D44cf12ee701C2503525], methods=[eth_sendTransaction, personal_sign, eth_sign, eth_signTypedData], events=[chainChanged, accountChanged], extensions=null)}, relayProtocol=null)`

via the `Web3Wallet.approveSession()` method, I have neither no active sessions `Web3Wallet.getListOfActiveSessions()` nor a error via the callback of  `Web3Wallet.approveSession()`

Currently on SDK `com.walletconnect:android-bom:1.4.0`

**Comment by CyrilNb** [comment]
Are there any breaking changes from 1.4.0 to 1.6.1?

**Comment by CyrilNb** [comment]
It's fixed. A new release is pending approval for the Play Store

**Comment by jakubuid** [comment]
There are not any breaking changes.
That's awesome!
Can't wait to have a look!
Thanks for the the collab 🤝

**Comment by kecson** [comment]
> It's fixed. A new release is pending approval for the Play Store

Is there a new version of Spot that solves this problem?

**Comment by jakubuid** [comment]
@kecson please report this issue in the FlutterSDK repo: https://github.com/WalletConnect/WalletConnectFlutterV2/issues

**Comment by CyrilNb** [comment]
@kecson yes there is a new version of Spot available on the Play store which fixed this problem :)

**Comment by kecson** [comment]
> @kecson yes there is a new version of Spot available on the Play store which fixed this problem :)

is there a new version of iOS Spot which fixed this problem?

**Comment by kecson** [comment]
> @kecson yes there is a new version of Spot available on the Play store which fixed this problem :)



> @kecson yes there is a new version of Spot available on the Play store which fixed this problem :)

I update Android Spot Wallet (4.75.3) from Google Play, Use Spot scan QrCode connect success. Bug Dapp deep link to Spot fail to connect .

**Comment by jakubuid** [comment]
Hey @CyrilNb, unfortunately, the deep link flow is still broken as you can is in the video below. By scanning QR code I was able to connect without any issues 👌 
Spot version: 4.75.3


https://user-images.githubusercontent.com/23366251/225850683-682e3f7d-da23-4965-af11-63c9031945b9.mp4

**Comment by jakubuid** [comment]
Hi @CyrilNb lmk if I can help in any way :)

**Comment by CyrilNb** [comment]
It's fixed now and will be included in the next update
